tests/Paysera/WalletApi/MapperTest.php

```diff
@@ -114,4 +126,152 @@ public function testDecodesPep()
         self::assertEquals('relationValue', $pepObj->getRelation());
         self::assertEquals('positionAValue', $pepObj->getPositions()[0]);
     }
+
+    /**
+     * @param $input
+     * @param $expected
+     * @return void
+     * @dataProvider decodeClientPermissionsToWalletDataProvider
+     */
+    public function testDecodeClientPermissionsToWallet($input, $expected)
+    {
+        self::assertEquals($expected, $this->mapper->decodeClientPermissionsToWallet($input));
+    }
+
+    /**
+     * @param $input
+     * @param $expected
+     * @return void
+     * @dataProvider encodeClientPermissionsToWalletDataProvider
+     * @throws Paysera_WalletApi_Exception_LogicException
+     */
+    public function testEncodeClientPermissionsToWallet($input, $expected)
+    {
+        if ($expected instanceof Exception) {
+            self::setExpectedException(get_class($expected), $expected->getMessage());
+        }
+        self::assertEquals($expected, $this->mapper->encodeClientPermissionsToWallet($input));
+    }
+
+    public function decodeClientPermissionsToWalletDataProvider()
+    {
+        return [
+            'case_1_empty_scopes' => [
+                'input' => [
+                    'id' => 1,
+                    'account' => 'EVP1',
+                    'scopes' => [],
+                ],
+                'expected' => Paysera_WalletApi_Entity_ClientPermissionsToWallet::create()
+                    ->setWalletId(1)
+                    ->setAccountNumber('EVP1')
+                    ->setScopes([]),
+            ],
+            'case_2_balance_only' => [
+                'input' => [
+                    'id' => 1,
+                    'account' => 'EVP1',
+                    'scopes' => [
+                        'balance'
+                    ],
+                ],
+                'expected' => Paysera_WalletApi_Entity_ClientPermissionsToWallet::create()
+                    ->setWalletId(1)
+                    ->setAccountNumber('EVP1')
+                    ->setScopes([Paysera_WalletApi_OAuth_Consumer::SCOPE_BALANCE]),
+            ],
+            'case_3_statements_only' => [
+                'input' => [
+                    'id' => 1,
+                    'account' => 'EVP1',
```

Comment:
if you manually enter values in each case, would be good to use it and change values in there.

Reply:
Done